### PR TITLE
allows for adding multiple owners for a project

### LINKED ISSUE #2673: Change API to allow adding an owner to a project.
**Description** At the moment this is not possible because in the past we decided that a project can have only one owner. An owner can be assigned when adding a new member to the project or changing the group for already existing member. ` POST /projects/{project_id}/users` `PUT /projects/{project_id}/users/{user_id}`

**Acceptance criteria**
- a user can be assigned to `owners` group when adding him to the project.
- a user can be assigned to `owners` group when editing/changing the group name for that user.
- the newly assigned `owner` can be removed from the project only if there is more than one owner.
- the old `owner` can be removed from the project only if there is more than one owner.

**Questions**
Can the user that created the project remove himself from it ? Yes, No, Yes if there is another owner ?

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: this pull adds additional control loop to RBAC controller, it
observes UserProjectBinding resources and adds appropriate OwnerReferences to a project.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2673

**Special notes for your reviewer**: for local testing use https://github.com/kubermatic/dashboard-v2/pull/1042

**Release note**:

```
allows assigning multiple owners to a project. 
```

### CODE REVIEW

File: api/pkg/controller/rbac/rbac_controller.go [hunk 1]
```diff
@@ -304,6 +311,34 @@ func (c *Controller) enqueueProjectResource(obj interface{}, gvr schema.GroupVer
...
+	for c.processProjectBindingNextItem() {
+	}
+}
+
+func (c *Controller) processProjectBindingNextItem() bool {
+	key, quit := c.projectBindingsQueue.Get()
+	if quit {
+		return false
+	}
+	defer c.projectBindingsQueue.Done(key)
```
Comment (maciaszczykm) [inline api/pkg/controller/rbac/rbac_controller.go:324]: Please check if it does not causes any issues (calling it after possible return).

Comment (p0lyn0mial) [reply api/pkg/controller/rbac/rbac_controller.go:324]: it doesn't, `quit` ends processing, for reference i've checked the sample controller which has the same flow https://github.com/kubernetes/sample-controller/blob/master/controller.go#L187

**Comment by maciaszczykm** [comment]
Right now everything seems to work, tested it locally with https://github.com/kubermatic/dashboard-v2/pull/1042. 

/approve

**Comment by maciaszczykm** [comment]
Feel free to cancel the hold.

/lgtm
/hold
/test pull-kubermatic-e2e-aws-coreos-1.13

**Comment by p0lyn0mial** [comment]
@maciaszczykm thanks for testing and reviewing.

**Pull request merged**